### PR TITLE
Properly validate whether tranactions are standard

### PR DESCRIPTION
**Pull request by KonradStaniec**
Add:
- new `btcstaking` library function to validate whether transactions are standard based on https://github.com/btcsuite/btcd/blob/ee68dc66a835bf2c9333f3d7b33791841f561c84/mempool/policy.go#L285
- use those function to validate unbonding / slashing transactions
- test suite to test new functions

### CODE REVIEW

File: btcstaking/staking_test.go [hunk 1]
```diff
@@ -413,3 +413,148 @@ func TestNotAllowFinalityProviderKeysAsCovenantKeys(t *testing.T) {
 	require.Error(t, err)
 	require.True(t, errors.Is(err, btcstaking.ErrDuplicatedKeyInScript))
 }
+
+func TestCheckPreSignedTxSanity(t *testing.T) {
+	tests := []struct {
```
Comment (Lazar955) [inline btcstaking/staking_test.go:418]: t.Parallel()

File: btcstaking/staking_test.go [hunk 2]
```diff
@@ -413,3 +413,148 @@ func TestNotAllowFinalityProviderKeysAsCovenantKeys(t *testing.T) {
...
+			numInputs:      1,
+			numOutputs:     1,
+			maxTxVersion:   2,
+			wantErr:        true,
+			expectedErrMsg: "tx weight must not exceed 400000",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := btcstaking.CheckPreSignedTxSanity(
```
Comment (Lazar955) [inline btcstaking/staking_test.go:548]: t.Parallel() in subtest also

**Review by Lazar955 — APPROVED** [review]
Nice work

**Review by SebastianElvis — APPROVED** [review]
Does this need to be backported or it can go to the next version?

File: btcstaking/staking.go [hunk 3]
```diff
@@ -215,6 +223,89 @@ func IsSimpleTransfer(tx *wire.MsgTx) error {
 	return nil
 }
 
+// CheckPreSignedTxSanity perfomorms basic checks on a pre-signed transaction:
```
Comment (gitferry) [inline btcstaking/staking.go:226]:
```suggestion
// CheckPreSignedTxSanity performs basic checks on a pre-signed transaction:
```

**Review by gitferry — APPROVED** [review]
Great work!

**Comment by KonradStaniec** [comment]
> Does this need to be backported or it can go to the next version?

I wanted to backport it to release patch version v0.12.1, and bump phase-1 services which use this validation: staking-api, and covenant-emulator.

Indexer does not need it as it reads data from BTC ledger, so if miner inserted v0 transaction into the ledger, then indexer must process it

**Pull request merged**